sdk/js/src/service/devices/index.js

```diff
@@ -513,6 +515,29 @@ class Devices {
         'end_device.ids.application_ids.application_id': applicationId,
       },
     }
+
+    // If CAC is set, attempt to claim the End Device via the DCS.
+    if (devicePayload.end_device.claim_authentication_code) {
+      try {
+        const payload = {
+          authenticated_identifiers: {
+            authentication_code: devicePayload.end_device.claim_authentication_code.value,
+          },
+          target_device_id: deviceId,
+          target_application_ids: {
+            application_id: applicationId,
+          },
+        }
+        const response = await this._api.EndDeviceClaimingServer.Claim(undefined, payload)
+
+        return Marshaler.payloadSingleResponse(response)
+      } catch (error) {
+        if (error.statusCode === 400) {
+          throw error
+        }
```

Comment:
I don't think we should only care about 400s here. If the claiming fails for whatever reason while having a CAC, we probably don't want to fall back to normal registration. I think it's legitimate to stop the process altogether then and throw the error.

@johanstokking any thoughts on this? Also, is it right to use only the CAC as an indicator for claiming, or should this be done more explicitly, so we can also allow manual registration even when setting a CAC?

Reply:
> If the claiming fails for whatever reason while having a CAC, we probably don't want to fall back to normal registration. I think it's legitimate to stop the process altogether then and throw the error.

ACK indeed. We should not support fallbacks for claimable devices. If rollback is tricky and we don't want to leave devices in a weird state, you can also first check if claiming succeeds before creating it in the NS/AS/IS.

>  Also, is it right to use only the CAC as an indicator for claiming, or should this be done more explicitly, so we can also allow manual registration even when setting a CAC?

I think a combination of checking the JoinEUI for claimability (`/api/v3/edcs/claim/info`) and if there is a CAC is the right way to go.